src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Messaging;
+using Microsoft.CmdPal.UI.ViewModels.Messages;
+
+namespace Microsoft.CmdPal.UI.ViewModels;
+
+public partial class AliasManager : ObservableObject
+{
+    private readonly TopLevelCommandManager _tlcManager;
+    private readonly Dictionary<string, CommandAlias> _aliases = [];
+
+    public AliasManager(TopLevelCommandManager tlcManager)
+    {
+        _tlcManager = tlcManager;
+        PopulateDefaultAliases();
+    }
+
+    private void AddAlias(CommandAlias a) => _aliases.Add(a.SearchPrefix, a);
+
+    public bool CheckAlias(string searchText)
+    {
+        if (_aliases.TryGetValue(searchText, out var alias))
+        {
+            try
+            {
+                var tlc = _tlcManager.LookupCommand(alias.CommandId);
+                if (tlc != null)
+                {
+                    WeakReferenceMessenger.Default.Send<ClearSearchMessage>();
+                    WeakReferenceMessenger.Default.Send<PerformCommandMessage>(new(new(tlc.Command)));
```

Comment:
Should these be linked somehow? I believe we have the search box tied to the page that's being displayed as navigation occurs. So if the command navigates, then that should be handled. Are there just other scenarios where when we're processing the PerformCommandMessage that the searchbox should be reacting to?

Reply:
Yea this felt odd as I was writing it. 

Aliases are a uniquely main page thing, and IMO it makes sense to clear out the main page's search text when going to an aliased command. That way, when you go back, you don't have `>` just sitting there waiting for you

For pretty much any other scenario though, I feel like you want to leave the previous page's search state intact, so that when you go back, it's still there. Like if I'm in the Obsidian notes list, and I filter to "Extensions", preview a note, then go back, I should still get "Extensions" in the search. 

So, it's only this weird wacky main page that acts like this.